test/telemetry/src/main/java/org/opensearch/test/telemetry/tracing/SpanDataValidator.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.test.telemetry.tracing;
+
+import org.opensearch.telemetry.tracing.Span;
+
+import java.util.List;
+
+/**
+ * Performs validations on span Data.
+ */
+public interface SpanDataValidator {
```

Comment:
Shall we rename it as `TracingValidator` instead as it validates at trace level instead of span level